darwin/datatypes.py

```diff
@@ -284,6 +284,53 @@ def full_path(self) -> str:
         return construct_full_path(self.remote_path, self.filename)
 
 
+@dataclass(frozen=True, eq=True)
+class ConversionError:
+    """
+    Represents a failed conversion to YOLO format.
```

Comment:
While I like `ConversionError`, I feel like it would be useful not just for YOLO. Not suggesting that we should use it in all export modules in this PR though.

Devil's advocate: do we _need_ it at all? We could return `None` if the conversions fail, and instead of keeping conversion errors in memory, use a file logger to keep track of all the errors. 🤔

Reply:
> Devil's advocate: do we need it at all? We could return None if the conversions fail, and instead of keeping conversion errors in memory, use a file logger to keep track of all the errors. thinking

I thought about that (discarding error info) but went against it after talking to Rafal Z. It turns out users really like to know _why_ something fails. An error in conversion is something potentially useful for a user to see after an operation completes.

If you still want a logger, we can still use this data structure for such a thing. I personally like the idea.

> While I like ConversionError, I feel like it would be useful not just for YOLO. Not suggesting that we should use it in all export modules in this PR though.

This is the direction I would like to move. Testing waters here, lets see how this PR/functionality goes and then draw conclusions / improvements :D